### DISCUSSION TITLE
Can't request emergency access

### DISCUSSION THREAD

**Discussion by Kr0kspluish** [dani-garcia/vaultwarden#3590] [Q&A] [answered]
### Subject of the issue
I have a user of my vaultwarden server who set me as emergency contact, with takeover level. 
I need to request emergency access to their vault now, I can see them in the Emergency Access panel of my account settings with the correct "Accepted" and 'Takeover" labels, but I can't make the request access as the button for doing so does not appear. I tried on different browsers (Firefox, Edge, Duckduckgo android).
I feel like I saw that button when we set the emergency access up months ago, so maybe it disappeared with an upgrade of the docker image?

### Deployment environment
* Vaultwarden version: v1.28.1
* Web-vault version: v2023.3.0b
* OS/Arch: linux/aarch64
* Running within Docker: true (Base: Debian)
* Environment settings overridden: false
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.39.2
* Clients used: web vault
* Reverse proxy and version: SWAG docker, version 2.4.0-ls195

### Config

**Environment settings which are overridden:** 

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "***************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://**************",
  "domain_origin": "*****://**************",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 2,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Vaultwarden des kiwis",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "*************************",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": false,
  "password_iterations": 100000,
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "*********************",
  "smtp_from_name": "Kimou - Vaultwarden",
  "smtp_host": "**************",
  "smtp_password": "***",
  "smtp_port": 587,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": "*********************",
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": 100,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": 1000000,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": true,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```

### Steps to reproduce
Create at least two users A and B and set up emergency access so A can access B's vault (takeover).
Go to account settings of A, and check whether the request access feature shows up.

### Expected behaviour
According to https://bitwarden.com/help/emergency-access/ (pointed to by the Vaultwarden wiki), a gear icon with a menu including "request access" should be there.

### Actual behaviour
It is not there; instead there is only the three vertical dots icon and the only options that appears when clicking it is "remove".

### Troubleshooting data

![Screenshot](https://github.com/dani-garcia/vaultwarden/assets/51377052/7991e3af-24cb-43fb-8961-e64a3ebbabe5)

**Comment by BlackDex** [comment]
And what does the other account show? Had the invite been excepted? And afterwards been confirmed? Without confirmation it will not work.

**Comment by Kr0kspluish** [comment]
I can't answer that, as the other user actually lost access to their vault and they needed me to recover it. 
So if the feature is not known to be broken, I must suppose they did not confirm the emergency contact set up and I apologize for wasting your time.

**Comment by BlackDex** [comment, marked as answer]
I can confirm it works as it should, and i can also see that the other end did not confirmed your acceptation.
There should be only one label there like so.

![image](https://github.com/dani-garcia/vaultwarden/assets/610450/f2a77a5f-ea91-403f-933b-c6c828753639)

The problem is, you can't even just change a flag in the database, since you need a security token, which only gets generated during the confirm action. That token provides you access to the other users vault. Without it, that is a no-go.
